README.md

```
@@ -176,13 +196,97 @@ spec:
 When a resource is excluded using this label, the plugin will not return it's address.
 
+## Headless Service Support
+
+`k8s_gateway` can resolve headless services (services with `ClusterIP: None`) to return all pod IPs from the associated EndpointSlices. This is useful for scenarios where you need to expose StatefulSet pods or other workloads with stable network identities via DNS.
+
+### Enabling Headless Service Resolution
+
+To enable headless service resolution:
+
+1. **Add the resource to your Corefile configuration:**
+   ```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)